keps/sig-cli/0015-kubectl-plugins.md

```diff
@@ -0,0 +1,151 @@
+---
+kep-number: 15
+title: Kubectl Plugins 
+authors:
+  - "@juanvallejo"
+  - "@soltysh"
+owning-sig: sig-cli
+participating-sigs:
+  - sig-cli
+reviewers:
+  - "@pwittrock"
+  - "@deads2k"
+  - "@liggitt"
+  - "@soltysh"
+approvers:
+  - "@pwittrock"
+  - "@soltysh"
+editor: juanvallejo
+creation-date: 2018-07-24
+last-updated: yyyy-mm-dd
+status: provisional
+see-also:
+  - n/a
+replaces:
+  - "https://github.com/kubernetes/community/blob/master/contributors/design-proposals/cli/kubectl-extension.md"
+  - "https://github.com/kubernetes/community/pull/481"
+superseded-by
+  - n/a
+---
+
+# Kubectl Plugins 
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Scenarios]](#scenarios)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+
+## Summary
+
+This proposal introduces the main design for a plugin mechanism in `kubectl`.
+The mechanism is a git-style system, that looks for executables on a user's `$PATH` whose name begins with `kubectl-`.
+This design allows plugin binaries to override specific command paths, replacing allowed subcommands in existing commands,
+or adding new subcommands under the `kubectl plugin` path.
+
+## Motivation
+
+The existing alpha plugin system in `kubectl` presents a few limitations with its current design. 
+It forces plugin scripts and executables to exist in a pre-determined location, requires basic information for a single 
+plugin to be declared explicitly in its own file, and does not provide a clear way to override existing command paths, 
+or provide additional subcommands without having to override a top-level command.
+
+A re-design plugins system allows us to implement extensibility requests from users that the current system cannot address
+(See https://github.com/kubernetes/kubernetes/issues/53640).
+
+### Goals
+
+* Support a simple plugin installation process (no additional config, users drop a binary in their `PATH`, for example, 
```

Comment:
We don't want to support any installation process, we just pick up what's available in the `$PATH`.